include/aspect/postprocess/global_statistics.h

```diff
@@ -0,0 +1,155 @@
+        void
+        parse_parameters (ParameterHandler &prm);
+
+
+      private:
+        /**
+         * Write global statistics such as the time step number and the
+         * number of degrees of freedom into the statistics object.
+         */
+        void
+        generate_global_statistics(TableHandler &statistics);
+
+        /**
+         * This function clears all the saved solver information that is
+         * stored in this class. It is executed every time the output is
+         * written into the statistics object, but also after each initial
+         * refinement step, in case it is not written (to avoid summing
+         * information across different refinement steps).
+         */
+        void
+        clear_data();
+
+        /**
+         * Callback function that is connected to the post_stokes_solver
+         * signal to store the solver history.
+         */
+        void
+        store_stokes_solver_history(const unsigned int number_S_iterations,
+                                    const unsigned int number_A_iterations,
+                                    const SolverControl &solver_control_cheap,
+                                    const SolverControl &solver_control_expensive);
+
+        /**
+         * Callback function that is connected to the post_advection_solver
+         * signal to store the solver history.
+         */
+        void
+        store_advection_solver_history(const bool solved_temperature_field,
+                                       const unsigned int compositional_index,
+                                       const SolverControl &solver_control);
+
+        /**
+         * Variables that store the Stokes solver history of the current
+         * timestep, until they are written into the statistics object
+         * upon the call to execute(). They are cleared after writing
+         * the content.
+         */
+        std::vector<unsigned int> list_of_S_iterations;
+        std::vector<unsigned int> list_of_A_iterations;
+        std::vector<SolverControl> solver_controls_cheap;
+        std::vector<SolverControl> solver_controls_expensive;
+
+        /**
+         * A container that stores the advection solver history of the current
+         * timestep, until it is written into the statistics object
+         * upon the call to execute(). It is cleared after writing
+         * the content. The vector contains pairs, which consist
+         * of a column name (for the temperature or one of the
+         * compositional fields), and a vector of SolverControl
+         * objects (one per nonlinear iteration for this particular
+         * field). This layout allows storing varying numbers of
+         * nonlinear iterations for temperature and compositional fields
+         * (if any nonlinear solver scheme would implement that at
+         * some point).
+         */
+        std::vector<std::pair<std::string, std::vector<SolverControl> > > advection_solver_controls;
```

Comment:
This is interesting -- why copy the whole `SolverControl` object into the vector? Wouldn't it be enough to extract the number of iterations, or a pair of numbers?